extensions/database/module/scripts/index/database-source-ui.js

```diff
@@ -454,21 +469,45 @@ Refine.DatabaseSourceUI.prototype._validateNewConnectionForm = function() {
         var initialDatabaseInput = $.trim(self._elmts.initialDatabaseInput[0].value);
         var initialSchemaInput = $.trim(self._elmts.initialSchemaInput[0].value);
         
-        if (databaseHostInput.length === 0) {
+        var alphaNumRE = /^[a-zA-Z0-9._-]*$/;
+        var numRE = /^[0-9]*$/;
+        
+        var alphaNumConnNameTestResult = alphaNumRE.test(connectionNameInput);
+        var databaseHostTestResult = alphaNumRE.test(databaseHostInput);
+        var databasePortTestResult = numRE.test(databasePortInput);
+        var databaseUserTestResult = alphaNumRE.test(databaseUserInput);
+        
+        if(alphaNumConnNameTestResult == false){
+        	window.alert($.i18n('database-source/alert-conn-name-invalid-character'));
+        	return false;
+        }else if(connectionNameInput.contains("<script>") || connectionNameInput.contains("<img>")){
+        	window.alert($.i18n('database-source/alert-conn-name-invalid-character'));
+        	return false;
+        }else if (databaseHostInput.length === 0) {
             window.alert($.i18n('database-source/alert-server'));
             return false;
+        }else if(databaseHostInput.contains("<script>") || databaseHostInput.contains("<img>")){
+        	window.alert($.i18n('database-source/alert-db-host-invalid-character'));
+        	return false;
```

Comment:
Rather than doing this validation by checking for dangerous HTML tags, I think it would be simpler to just make sure the connection name (and other parameters) are HTML-escaped before being rendered to the DOM. This way, the user cannot insert any HTML tags.

Reply:
This is just an extra check. The field will accept only Alphanumeric, period, underscore and hyphen only. I just added that line for extra check. As for html-escaping in the frontend, I'll have to check the javascript code again.